src/facade/cmd_arg_parser.h

```diff
@@ -0,0 +1,232 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/strings/ascii.h>
+#include <absl/strings/match.h>
+
+#include <optional>
+#include <string_view>
+
+#include "base/logging.h"
+#include "facade/error.h"
+#include "facade/facade_types.h"
+
+namespace facade {
+
+// Utility class for easily parsing command options from argument lists.
+struct CmdArgParser {
```

Comment:
Is it actually ok to keep this header only?

Reply:
ah, probably not (noticing 2s after giving the approval)